examples/NativeExamples/user_trace_003_no_predicates.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// This example doing the same as user_trace_002 but using native events filtering (not a predicates)
+//
+
+#include <iostream>
+#include <cassert>
+
+#include "..\..\krabs\krabs.hpp"
+#include "examples.h"
+
+void user_trace_003_no_predicates::start()
+{
+    // user_trace instances should be used for any non-kernel traces that are defined
+    // by components or programs in Windows. They can optionally take a name -- if none
+    // is provided, a random GUID is assigned as the name.
+    krabs::user_trace trace(L"My Named Trace");
+
+    // A trace can have any number of providers, which are identified by GUID. These
+    // GUIDs are defined by the components that emit events, and their GUIDs can
+    // usually be found with various ETW tools (like wevutil).
+
+	//listen for file events
+    krabs::provider<> provider(krabs::guid(L"{EDD08927-9CC4-4E65-B970-C2560FB5C289}"));
+
+    // In user_trace_001.cpp, we manually filter events by checking the event information
+    // in our callback functions. In this example, we're going to use a provider filter
+    // to do this for us.
+
+    // We instantiate an event_filter first. An event_filter is created with a
+    // predicate -- literally just a function that does some check on an EVENT_RECORD
+    // and returns a boolean -- true when the event should be passed on to callbacks,
+    // and false otherwise.
+
+    // krabs provides direct native filtering - that will boost performance and not liesten 
+	//for event which not required. We'll use one of those to filter based on the event id.
+
+	//listen for file delete event
```

Comment:
Nit: please align the new comments with the preceding ones. There is also a typo above (`liesten`). You may want to say that krabs `supports provider filtering` rather than `provides direct native filtering` since it's up to the caller to choose which one he/she will use.

Referring to this as "provider filtering" rather than "native filtering" aligns it with Microsoft's ETW API documentation .

Reply:
Done